### ISSUE TITLE
[webgpu]exception with a weired number only

### ISSUE THREAD

**Issue by vacing** [microsoft/onnxruntime#16267]
### Describe the issue

test with minist model, got weired exception, has no stack info but a weired number only
webgl xnnpack wasm all is ok except webgpu
```
[V,2023-06-07T08:21:29.340Z][WebGPU] GpuDataManager.create(size=16) => id=0
onnx.html:87 Uncaught (in promise) 8314224
```

### To reproduce

[Desktop.zip](https://github.com/microsoft/onnxruntime/files/11674394/Desktop.zip)
Chrome: 114.0.5735.110

### ONNX Runtime Installation

Released Package

### ONNX Runtime Version or Commit ID

release version

### Execution Provider

'webgpu' (WebGPU)

**Comment by guschmue** [comment]
we fixed this recently - works on newer dev builds and main.
You could try a build from here:
```
<script src="https://cdn.jsdelivr.net/npm/onnxruntime-web@1.16.0-dev.20230606-f013965831/dist/ort.webgpu.min.js" crossorigin></script>
```

**Comment by vacing** [comment]
> we fixed this recently - works on newer dev builds and main. You could try a build from here:
> 
> ```
> <script src="https://cdn.jsdelivr.net/npm/onnxruntime-web@1.16.0-dev.20230606-f013965831/dist/ort.webgpu.min.js" crossorigin></script>
> ```

yes, it works for this case
when I test another [case](https://github.com/microsoft/onnxruntime/issues/15796#issuecomment-1577328476), it has no error   but the page is corrupted
> chrome  114.0.5735.110

**Comment by vacing** [comment]
when I test [this model](https://github.com/creativeIKEP/SelfieSegmentationBarracuda/blob/main/Packages/SelfieSegmentationBarracuda/ONNX/selfie_segmentation.onnx), I get the following error

```
![image](https://github.com/microsoft/onnxruntime/assets/6896966/982ff397-b25a-4044-b176-4130dbb24bfb)

```

**Comment by gabrielgrant** [comment]
looks like the original issue was fixed in builds beyond 1.16.0-dev.20230606-f013965831

And am I understanding correctly the [remaining](https://github.com/microsoft/onnxruntime/issues/16267#issuecomment-1581776928) [issues](https://github.com/microsoft/onnxruntime/issues/16267#issuecomment-1581794335) should be fixed by #16393 ?

**Comment by guschmue** [comment]
I assume this is fixed.
